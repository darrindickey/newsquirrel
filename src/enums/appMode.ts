export enum AppMode {
  LIGHT=1,
  DARK=2,
}