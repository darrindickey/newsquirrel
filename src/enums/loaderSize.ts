enum LoaderSize {
  SMALL,
  LARGE
}

export default LoaderSize;