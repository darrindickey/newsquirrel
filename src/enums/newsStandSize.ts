enum NewsStandSize {
  COMPACT,
  COZY,
  IMAGE_FREE
}

export default NewsStandSize;