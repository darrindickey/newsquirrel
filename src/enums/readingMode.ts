enum ReadingMode {
  TOP_HEADLINES,
  SEARCH_EVERYTHING
}

export default ReadingMode;