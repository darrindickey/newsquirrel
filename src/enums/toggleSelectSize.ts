enum toggleSelectSize {
  SMALL,
  LARGE
}

export default toggleSelectSize;