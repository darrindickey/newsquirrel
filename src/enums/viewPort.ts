
enum ViewPort {
  MOB,
  DESK
}

export default ViewPort;
