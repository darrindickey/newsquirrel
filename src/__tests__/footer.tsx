import 'jest-dom/extend-expect';
import * as React from 'react';
import {render} from 'react-testing-library';
import Footer from '../components/footer/footer';
import { AppMode } from '../enums/appMode';

test('Footer renders correctly', () => {
  const {getByTestId} = render(<Footer appMode={{name: 'Dark', value:AppMode.DARK}}/>);
  const node = getByTestId('rt-footer');
  const link = node.querySelector('a');
  expect(node).toBeInTheDocument();
  expect(node).toMatchSnapshot();
})

