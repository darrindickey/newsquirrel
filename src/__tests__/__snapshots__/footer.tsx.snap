// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  class="sc-bdVaJa jqquVI"
  data-testid="rt-footer"
>
  <div
    class="sc-bwzfXH hmXunB"
  >
    <a
      href="https://twitter.com/prabhumurthy2"
      target="new"
    >
      <img
        class="sc-htpNat ewHMNg"
        height="20"
        src="https://cdn.jsdelivr.net/npm/simple-icons@latest/icons/twitter.svg"
        width="20"
      />
    </a>
    <a
      href="https://github.com/prabhuignoto"
      target="new"
    >
      <img
        class="sc-htpNat ewHMNg"
        height="20"
        src="https://cdn.jsdelivr.net/npm/simple-icons@latest/icons/github.svg"
        width="20"
      />
    </a>
  </div>
  <div
    class="sc-ifAKCX WOpjk"
  >
    <span>
      2018 Â© Prabhu Murthy
    </span>
  </div>
  <div
    class="sc-EHOje fhxHOS"
  >
    Designed & developed with ðŸ–¤ by me
  </div>
  <div
    class="sc-bxivhb cDFCAd"
  >
    <a
      href="https://newsapi.org"
    >
      Powered by News API
    </a>
  </div>
</footer>
`;
